code/modules/multiz/hoist.dm

```diff
@@ -0,0 +1,270 @@
+///////////////////////////
+// Dost thou even hoist? //
+///////////////////////////
+
+#define NORMAL_LAYER 3
+
+/obj/item/hoist_kit
+	name = "hoist kit"
+	desc = "A setup kit for a hoist that can be used to lift things. The hoist will deploy in the direction you're facing."
+	icon = 'icons/obj/hoists.dmi'
+	icon_state = "hoist_case"
+
+/obj/item/hoist_kit/attack_self(mob/user)
+	new /obj/structure/hoist (get_turf(user), user.dir)
+	user.visible_message(span("warning", "[user] deploys the hoist kit!"), span("notice", "You deploy the hoist kit!"), span("notice", "You hear the sound of parts snapping into place."))
+	qdel(src)
+
+/obj/effect/hoist_hook
+	name = "hoist clamp"
+	desc = "A clamp used to lift people or things."
+	icon = 'icons/obj/hoists.dmi'
+	icon_state = "hoist_hook"
+	var/obj/structure/hoist/source_hoist
+	can_buckle = 1
+	anchored = 1
+	description_info = "Click and drag someone (or any object) to this to attach them to the clamp. If you are within reach, when you click and drag this to a turf adjacent to you, it will move the attached object there and release it."
+
+/obj/effect/hoist_hook/attack_hand(mob/living/user)
+	return // This has to be overridden so that it works properly.
+
+/obj/effect/hoist_hook/MouseDrop_T(atom/movable/AM,mob/user)
+	if (use_check(user, USE_DISALLOW_SILICONS))
```

Comment:
In this PR at least, `use_check()` it being used only once: here.  
This line could have been replaced with  `if(issilicon(user))`  instead of porting use_check.    

The rest of use_check's functionality is unused.  There are even places only a few lines later where it could have been used but isn't!  
That said, I **do** like the idea of use_check in principle, its kind of cool to consolidate some of these checks in a proc where it can be standardized!  If we actually use it...

Reply:
It's all used, actually. It checks for certain flags *not* being present, and assumes a default use case.